driver-core/src/main/java/com/datastax/driver/core/querybuilder/QueryBuilder.java

```diff
@@ -351,6 +351,19 @@ public static Clause in(String name, Object... values) {
      * @param values the values
      * @return the corresponding where clause.
      */
+    public static Clause in(String name, Collection<?> values) {
+        return new Clause.InClause(name, values);
+    }
+
+    /**
+     * Create an "in" {@code WHERE} clause stating the provided column must be equal
+     * to one of the provided values.
+     *
+     * @param name   the column name
+     * @param values the values
+     * @return the corresponding where clause.
+     * @deprecated Use {@link #in(String, Collection)} instead.
+     */
     public static Clause in(String name, List<?> values) {
```

Comment:
Kind of annoying though to deprecate because people using a list are going to have to manually cast their list to `Collection` in order to not use a deprecated method and avoid a warning. Imo we could remove it and add a clirr exception since the `QueryBuilder` class is final and therefore should not be extended. It would also not impact users just using because List is a child of Collection.

Reply:
D'oh, good catch.  

> Kind of annoying though to deprecate because people using a list are going to have to manually cast their list to Collection in order to not use a deprecated method and avoid a warning. Imo we could remove it and add a clirr exception since the QueryBuilder class is final and therefore should not be extended. It would also not impact users just using because List is a child of Collection.

I think getting a compiler warning is a better tradeoff than possibly breaking user code.  In general case it won't, but if a user is using reflection or some other high level language there could be unknown consequences of changing the method signature.

Reply:
Hmm I can't say about higher level tools that use reflection and so on on which it would potentially break so I don't know how common it is. However deprecating would lead _every_ existing user to use a deprecated method and get a warning, and lead all of them to do that `(Collection<?>)` casting to avoid it. I'd rather affect less people but would be ok to go the "deprecation" way